src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/UserAttemptManager.java

```diff
@@ -0,0 +1,111 @@
+        List<ContentSummaryDTO> relatedContentSummaries = content.getRelatedContent();
+        if (relatedContentSummaries != null) {
+            for (ContentSummaryDTO relatedContentSummary : relatedContentSummaries) {
+                augmentContentSummaryWithAttemptInformation(relatedContentSummary, usersQuestionAttempts);
+            }
+        }
+        // for all children recurse
+        List<ContentBaseDTO> children = content.getChildren();
+        if (children != null) {
+            for (ContentBaseDTO child : children) {
+                if (child instanceof ContentDTO) {
+                    ContentDTO childContent = (ContentDTO) child;
+                    augmentRelatedQuestionsWithAttemptInformation(childContent, usersQuestionAttempts);
+                }
+            }
+        }
+    }
+
+    private void augmentContentSummaryWithAttemptInformation(
+            final ContentSummaryDTO contentSummary,
+            final Map<String, ? extends Map<String, ? extends List<? extends LightweightQuestionValidationResponse>>> usersQuestionAttempts) {
+        String questionId = contentSummary.getId();
+        Map<String, ? extends List<? extends LightweightQuestionValidationResponse>> questionAttempts = usersQuestionAttempts.get(questionId);
+        boolean questionAnsweredCorrectly = false;
+        if (questionAttempts != null) {
+            for (String relatedQuestionPartId : contentSummary.getQuestionPartIds()) {
+                questionAnsweredCorrectly = false;
+                List<? extends LightweightQuestionValidationResponse> questionPartAttempts = questionAttempts.get(relatedQuestionPartId);
+                if (questionPartAttempts != null) {
+                    for (LightweightQuestionValidationResponse partAttempt : questionPartAttempts) {
+                        questionAnsweredCorrectly = partAttempt.isCorrect();
+                        if (questionAnsweredCorrectly) {
+                            break; // exit on first correct attempt
+                        }
+                    }
+                }
+                if (!questionAnsweredCorrectly) {
+                    break; // exit on first false question part
+                }
+            }
+        }
+        contentSummary.setCorrect(questionAnsweredCorrectly);
+    }
+
+    /**
+     * Augment a list of content summary objects with user attempt information.
+     * @param user the user to augment the content summary list for.
+     * @param summarizedResults the content summary list to augment.
+     * @param hideCompleted whether to filter out completed content.
+     * @return the augmented content summary list.
+     * @throws SegueDatabaseException if there is an error retrieving question attempts.
+     */
+    public List<ContentSummaryDTO> augmentContentSummaryListWithAttemptInformation(RegisteredUserDTO user, List<ContentSummaryDTO> summarizedResults, boolean hideCompleted) throws SegueDatabaseException {
```

Comment:
The filtering feels a bit odd to be inside this function, since it is not obviously related to the rest of the method and can be (and is) done separately after the function with the results.
I think it is fine for now, though!